src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs

```diff
@@ -436,8 +436,8 @@ internal IQueryable ApplyPaging(IQueryable result, ODataQuerySettings querySetti
             if (pageSize > 0)
             {
                 bool resultsLimited;
-                result = LimitResults(result, pageSize, querySettings.EnableConstantParameterization, out resultsLimited);
-                if (resultsLimited && InternalRequest.RequestUri != null && InternalRequest.RequestUri.IsAbsoluteUri &&
+                result = LimitResults(result, pageSize, out resultsLimited);
```

Comment:
@KanishManuja-MS:Don't you want to retain the querySettings.EnableConstantParameterization parameter?